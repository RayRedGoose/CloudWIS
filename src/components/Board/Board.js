import React from 'react'
import './Board.scss'
import Menu from '../menu/Menu/Menu'
import WeatherWidget from '../widget/WeatherWidget/WeatherWidget'

const Board = (props) => {
  return (
    <div className="board-wrapper">
      <Menu
        changeCity={props.changeCity}
        searchHistory={props.searchHistory}
        fetchWeatherData={props.fetchWeatherData}/>
      <WeatherWidget
        searchCity={props.currentCity}
        weatherCurrentData={props.weatherData.today}
        weatherDaysData={props.weatherData.week} />
    </div>
  )
}

export default Board
