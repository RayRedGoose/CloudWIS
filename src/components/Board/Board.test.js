import React from 'react'
import { shallow } from 'enzyme'
import Board from './Board'

it('should match the snapshot with all data passed in correctly', () => {
  const mockProps = {
    currentCity: 'Denver',
    weatherData: {
      today: {desc: 'today weather'},
      week: {desc: 'week weather'}
    },
    searchHistory: ['Denver', 'Phoenix'],
    changeCity: jest.fn(),
    fetchWeatherData: jest.fn()
  }

  const board = shallow(<Board {...mockProps} />)
  expect(board).toMatchSnapshot()
})
