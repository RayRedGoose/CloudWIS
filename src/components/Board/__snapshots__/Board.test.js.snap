// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with all data passed in correctly 1`] = `
<div
  className="board-wrapper"
>
  <Menu
    changeCity={[MockFunction]}
    fetchWeatherData={[MockFunction]}
    searchHistory={
      Array [
        "Denver",
        "Phoenix",
      ]
    }
  />
  <WeatherWidget
    searchCity="Denver"
    weatherCurrentData={
      Object {
        "desc": "today weather",
      }
    }
    weatherDaysData={
      Object {
        "desc": "week weather",
      }
    }
  />
</div>
`;
